draft-ietf-ppm-dap.md

```diff
@@ -1222,6 +1250,9 @@ following generic checks.
 1. Check if the report's timestamp has passed its task's `task_expiration` time,
    if so the Aggregator MAY mark the input share as invalid with the error
    "task_expired".
+1. Check if the Report contains unrecognized extensions, either in
+   ReportMetadata.extensions or PlaintextInputShare.extensions. If so, then the
+   Aggregator MUST abort with "unrecognized_extension".
```

Comment:
Same observation here that I had about the problem document types: it's not clear to me how the leader will handle this differently than some more generic error code.

Reply:
Here I have a different view: Knowing what went wrong is useful telemetry because if we observe a spike in rejection rates, we'll want to investigate what's going on.

Reply:
Yes, I buy that we need a way for implementations to diagnose problems. In this case, you're arguing that you want the leader to be able to tell why a report share was rejected by the helper? This is handled reasonably gracefully anywhere we use HTTP problem documents because the `details` field allows adding arbitrary information for use in logging or metrics, but here we only have a single error code to insert in the list of `PrepareStep` structures. Maybe we could change `ReportShareError` so that instead of being just an enum, it's a structure that contains a value of the current `ReportShareError` enum and a `details` field that can optionally provide an implementation specific integer error code? Then the protocol wouldn't have to say anything specific about unrecognized extensions but your leader could still get something to track metrics on, and then ask your peer operating the helper why you're suddenly getting so many errors with `details = 18674749`, or whatever error code their implementation assigns to this case.

Reply:
Hmmm, that might be OK. I'm sort of used to enumerating different error types, based on my experience with TLS. (See ["AlertDescription"](https://datatracker.ietf.org/doc/html/rfc8446#section-6). Though I take your point that we should do our best minimize these.

One idea is to change "unrecognized_extension" to "unrecognized_message" and let this be a catch-all for messages that could not be handled ((to match our decision for upload flow). This is slightly odd, though, because there are cases where the Helper would not abort with "unrecognizedMessage" but would reject a report with "unrecognized_message".

Reply:
The precedent of TLS 1.3 does seem like a compelling argument, but I wonder if there were specific use cases for each of those errors.

I'll propose this as a way forward for this change:

- In this PR, use the generic `unrecognized_message` error. The principle here is to do the simplest thing that solves the problem we have, and to keep this PR focused on its problem statement.
- File an issue to extend `ReportShareError` as I suggest to allow implementation-specific error information in the aggregate sub-protocol.
- File an issue about whether we should use more specific error codes, and do more research on whether it's a good idea. We could for instance reach out to the TLS WG to see how they feel in retrospect about the error codes in TLS 1.3.

Reply:
I took the suggestion to use "unrecognized_message" here ... we can file issues for follow-up things once this PR merges.